python/GafferUI/LayoutMenu.py

```diff
@@ -86,6 +88,10 @@ def saveAs( name, menu ) :
 	scriptWindow, layouts = __scriptWindowAndLayouts( menu )
 
 	layout = scriptWindow.getLayout()
+
+	if hasattr( layout, 'updatePreferredBound' ) :
```

Comment:
Is this check here because in theory, we might have layouts that aren't CompoundEditors? The `ScriptWindow.setLayout()` method does name the argument `compoundEditor`, so unless I'm missing some other usage, I'd be in favour of just asserting the expected type in `setLayout()` and removing the checks. This would be in line with the eventual CompoundEditor->Layout rename that we talked about a bit.

Reply:
Yeah - some of the tests just shove a `PythonEditor` in there... so seen as it was a tested pattern, in the wild, seemed like other people may do this in esoteric Gaffer integrations - when they don't want a Layout, just an Editor?

Reply:
Managed to remove this in the end in https://github.com/GafferHQ/gaffer/pull/3167/commits/4361ca224e657be1db373d6d0fe91d18a252c235 as (since the call site has moved from where it originally was) that case shouldn't come up any more.